core/http/src/uri/absolute.rs

```diff
@@ -460,3 +495,44 @@ impl std::fmt::Display for Absolute<'_> {
         Ok(())
     }
 }
+
+#[cfg(feature = "serde")]
+mod serde {
+    use std::fmt;
+
+    use super::Absolute;
+    use _serde::{ser::{Serialize, Serializer}, de::{Deserialize, Deserializer, Error, Visitor}};
+
+    impl<'a> Serialize for Absolute<'a> {
+        fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+            serializer.serialize_str(&self.to_string())
+        }
+    }
+
+    struct AbsoluteVistor;
+
+    impl<'a> Visitor<'a> for AbsoluteVistor {
+        type Value = Absolute<'a>;
+        fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+            write!(formatter, "Expecting a valid URI string")
```

Comment:
This should say something like "an absolute URI"; no "expecting", lowercase start, and we don't need "valid". The other expecting strings should be fixed accordingly.